### PR TITLE
Analyze using GitLab CI

### PR DESCRIPTION
**Pull request by tomasz-kaminski-sonarsource**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+variables:
+  SONAR_SERVER_URL: "https://sonarcloud.io" 
+  SONAR_SCANNER_VERSION: 4.6.1.2450 # Find the latest version in the "Linux" link on this page:
+                                    # https://sonarcloud.io/documentation/analysis/scan/sonarscanner/ 
+  BUILD_WRAPPER_OUT_DIR: build_wrapper_output_directory # Directory where build-wrapper output will be placed
+  
+  SONAR_USER_HOME: "${CI_PROJECT_DIR}/.sonar"  # Defines the location of the analysis task cache
+  GIT_DEPTH: "0"  # Tells git to fetch all the branches of the project, required by the analysis task
+  # note that  SONAR_TOKEN are transmitted to the environament through Gitlab CI
```
Comment (arseniy-sonar) [inline .gitlab-ci.yml]:
```suggestion
  # note that SONAR_TOKEN is transmitted to the environment through Gitlab CI
```

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+get-sonar-binaries:
+  # this job sdownloads ans unpacks the build-wrapper and the sonar-scanner
```
Comment (arseniy-sonar) [inline .gitlab-ci.yml]:
```suggestion
  # this job downloads and unpacks the build-wrapper and the sonar-scanner
```

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  # in this example it is done for every build.
+  # This can be optimized by caching the files or better, by including them, in the build docker image
```
Comment (arseniy-sonar) [inline .gitlab-ci.yml]:
```suggestion
  # This can be optimized by caching the files or better, by including them in the build docker image.
```

**Review by arseniy-sonar — APPROVED** [review]
LGTM module a few typos and grammar corrections

File: README.adoc [hunk 4]
```diff
@@ -1,3 +1,41 @@
+= C++ example project scanned on SonarCloud using GitLab CI
+// URIs:
+:uri-qg-status: https://sonarcloud.io/dashboard?id=sc-sq-c-family-examples_linux-cmake-gitlab-ci-sc
+:img-qg-status: https://sonarcloud.io/api/project_badges/measure?project=sc-sq-c-family-examples_linux-cmake-gitlab-ci-sc&metric=alert_status
+:uri-build-status: https://gitlab.com/tomasz.kaminski/linux-cmake-gitlab-ci-sc/-/commits/main
+:img-build-status: https://gitlab.com/tomasz.kaminski/linux-cmake-gitlab-ci-sc/badges/main/pipeline.svg
+
+image:{img-build-status}[Build Status, link={uri-build-status}]
```
Comment (arseniy-sonar) [inline README.adoc:8]: Looks like this link and image are private. At least I do not have access to them (unauthenticated).

Comment (tomasz-kaminski-sonarsource) [reply README.adoc:8]: Yup there are also not visible for me after logout? Should I just remove them?

Comment (arseniy-sonar) [reply README.adoc:8]: Can you make the project public instead?
Just like this github project

Comment (tomasz-kaminski-sonarsource) [reply README.adoc:8]: Ok, this was caused by my attempt to rename the project to remove `s` from the name, the links still need to have `-scs` which will be confusing. I will recreate project.

**Pull request merged**